spm/spm_server.c

```diff
@@ -533,3 +550,33 @@ void psa_set_rhandle(psa_handle_t msg_handle, void *rhandle)
     spm_active_msg_t *active_msg = get_msg_from_handle(msg_handle);
     active_msg->channel->rhandle = rhandle;
 }
+
+void psa_eoi(uint32_t irq_signal)
+{
+    spm_partition_t *curr_partition = get_active_partition();
+    SPM_ASSERT(NULL != curr_partition);
+    if (curr_partition->irq_mapper == NULL) {
+        SPM_PANIC("Try to clear an interrupt flag without declaring IRQ");
+    }
+
+    if (0 == (curr_partition->flags_interrupts & irq_signal)) {
+        SPM_PANIC("Signal %d not in irq range\n", irq_signal);
+    }
+
+    bool is_one_bit = ((irq_signal != 0) && !(irq_signal & (irq_signal - 1)));
+    if (!is_one_bit) {
+        SPM_PANIC("signal 0x%x must have only 1 bit ON!\n",irq_signal);
+    }
+
+    IRQn_Type irq_line = curr_partition->irq_mapper(irq_signal);
```

Comment:
What if irq_mapper returns 0? Should we handle it?

Reply:
irq_mapper will always return a valid value, if an incorrect signal is requested we panic